pkg/api/vlabs/validate.go

```diff
@@ -354,6 +354,59 @@ func (a *Properties) validateMasterProfile() error {
 			return err
 		}
 	}
+
+	if m.IsVirtualMachineScaleSets() && a.OrchestratorProfile.OrchestratorType == Kubernetes {
```

Comment:
These blocks share a lot of common code with AgentPooProfile validation. Can we move them to a method and make sure they are re-used by MasterProfiles and AgentPoolProfiles?